### PR TITLE
fix: localStorage availability check update

### PR DESCRIPTION
**Pull request by iuriizaporozhets**
When in Safari set Block All cookies then local storage also will not work. But window.localStorage object itself will be available and when we will try set or get something from local storage we will get `SecurityError: The operation is insecure.` error and it is crashing the application for users with disabled cookies and local storage.

### CODE REVIEW

File: packages/nuqs/src/utils.ts [hunk 1]
```diff
@@ -38,3 +38,22 @@ export function getDefaultThrottle() {
...
   }
 }
+
+/**
+ * Check if localStorage is available
+ */
+export function isLocalStorageAvailable(){
+  try {
+    const test = 'test';
+
```
Comment (franky47) [inline packages/nuqs/src/utils.ts:48]: **formatting**: could you remove the extra empty lines between the code please?

File: packages/nuqs/src/utils.ts [hunk 2]
```diff
@@ -38,3 +38,22 @@ export function getDefaultThrottle() {
     return 320
   }
 }
+
+/**
+ * Check if localStorage is available
+ */
+export function isLocalStorageAvailable(){
+  try {
+    const test = 'test';
```
Comment (franky47) [inline packages/nuqs/src/utils.ts:47]: **suggestion**: make the key less prone to collisions, for example by marking it specifically for nuqs: `nuqs-localStorage-test`.

**Review by franky47 — CHANGES_REQUESTED** [review]
Thanks for this!

I see an issue with the current way it's implemented: if a client has a localStorage value under the `test` key, it would get cleared by what should otherwise be an idempotent function.

One alternative would be to read the value before attempting a write, or use a more specifically-named key, which won't cause collisions.

File: packages/nuqs/src/utils.ts (hunk 2 above)
Comment (iuriizaporozhets) [reply packages/nuqs/src/utils.ts]: @franky47 updated to `nuqs-localStorage-test` as suggested, thanks

**Review by franky47 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**